### ISSUE TITLE
Cannot access member "X" and "y" for type "Split"

### ISSUE THREAD

**Issue by Saif807380** [preprocessy/preprocessy#147]
In the `train_test_split` stage, when the number of rows in `train_df` and `train_y` are unequal, an error should be thrown. However, the line that throws the error, crashes the program with the error `Cannot access member 'X' and 'y' for type 'Split'. Unknown members.`

Expected Behaviour

Validate input should throw the following error `Number of samples of target label and feature dataframe unequal`.

Ref - https://github.com/preprocessy/preprocessy/blob/d8d1b1bb936d52fac4bfa97a3de0ebadda8741ea/preprocessy/data_splitting/_split.py#L78

Environment:

- Python version: Any
- Preprocessy version: Any
- Jupyter Notebook or Python Script: Both
